src/OpenSage.Game/Scene3D.cs

```diff
@@ -180,7 +180,7 @@ internal void LocalLogicTick(in TimeInterval gameTime, float tickT)
         {
             _orderGeneratorInputHandler?.Update();
 
-            foreach (var gameObject in GameObjects.Items)
+            foreach (var gameObject in GameObjects.Items.ToList())
```

Comment:
This is because there is a runtime error when the production system adds a new GameObject to the collection

Reply:
Ah, good point. Maybe we should instead add objects to a different list, and move them to the main collection after the loop is finished. This means new objects only appear at the start of the next frame, but I wouldn't be surprised if that's how the original engine works too.

Reply:
I've changed it to just use an array accessor instead of a foreach. Seems to work quite well, and it's cleaner than passing some kind of temporary list down the line :)